src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -59,6 +60,9 @@ def __init__(
         self._make_install_req_from_spec = make_install_req
         self._force_reinstall = force_reinstall
         self._ignore_requires_python = ignore_requires_python
+        self._upgrade_strategy = upgrade_strategy
```

Comment:
The legacy resolver has an assertion to make sure we pass the right value in. Maybe we should do the same here?

Reply:
+1. I'll add this tomorrow.